### PR TITLE
fixing weird exception throws when encountering an http client exception

### PR DESCRIPTION
**Pull request by slorello89**
_Describe your changes here_

## Contribution Checklist
* [x] Unit tests!
* [x] Updated [CHANGELOG.md](CHANGELOG.md)
* [x] My name is in [CONTRIBUTORS.md](CONTRIBUTORS.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,9 +4,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [5.2.2]
+### Fixed
+- Weird typed exceptions being thrown when there are apparent networking issues
```
Comment (dragonmantank) [inline CHANGELOG.md]: Probably shouldn't call this "weird", should reword this to "Better handling of exceptions [...]"

**Comment by dragonmantank** [comment]
LGTM :+1:

**Comment by lornajane** [comment]
Thanks for sorting this out! Are we stuck in approved-but-not-merged for a reason though? Do you need me to do something?
